### PR TITLE
Change Password Page Added for MVC

### PR DESCRIPTION
**Pull request by iyilm4z**
Fixes [this](https://github.com/aspnetboilerplate/aspnetboilerplate/issues/5249)

### CODE REVIEW

File: aspnet-core/src/AbpCompanyName.AbpProjectName.Web.Mvc/Views/Users/ChangePassword.cshtml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    <div class="container-fluid">
+        <div class="row">
+            <div class="col-12">
+                <form asp-route="ChangePassword" id="ChangePassword" method="post" autocomplete="off" class="form-horizontal">
+                    <div class="card">
+                        <div class="card-body">
+                            <div class="form-group row required">
+                                <label class="col-md-3 col-form-label">@L("CurrentPassword")</label>
+                                <div class="col-md-9">
+                                    <input type="text" class="form-control" name="CurrentPassword" id="CurrentPassword" required maxlength="@AbpUserBase.MaxPlainPasswordLength">
```
Comment (demirmusa) [inline aspnet-core/src/AbpCompanyName.AbpProjectName.Web.Mvc/Views/Users/ChangePassword.cshtml]:
```suggestion
                                    <input type="password" class="form-control" name="CurrentPassword" id="CurrentPassword" required maxlength="@AbpUserBase.MaxPlainPasswordLength">
```

File: aspnet-core/src/AbpCompanyName.AbpProjectName.Web.Mvc/Views/Users/ChangePassword.cshtml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                                </div>
+                            </div>
+                            <div class="form-group row required">
+                                <label class="col-md-3 col-form-label">@L("Password")</label>
+                                <div class="col-md-9">
+                                    <input type="text" class="form-control" name="NewPassword" id="NewPassword" required maxlength="@AbpUserBase.MaxPlainPasswordLength">
```
Comment (demirmusa) [inline aspnet-core/src/AbpCompanyName.AbpProjectName.Web.Mvc/Views/Users/ChangePassword.cshtml]:
```suggestion
                                    <input type="password" class="form-control" name="NewPassword" id="NewPassword" required maxlength="@AbpUserBase.MaxPlainPasswordLength">
```

File: aspnet-core/src/AbpCompanyName.AbpProjectName.Web.Mvc/Views/Users/ChangePassword.cshtml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                                </div>
+                            </div>
+                            <div class="form-group row required">
+                                <label class="col-md-3 col-form-label">@L("ConfirmPassword")</label>
+                                <div class="col-md-9">
+                                    <input type="text" class="form-control" name="ConfirmNewPassword" id="ConfirmNewPassword" required maxlength="@AbpUserBase.MaxPlainPasswordLength">
```
Comment (demirmusa) [inline aspnet-core/src/AbpCompanyName.AbpProjectName.Web.Mvc/Views/Users/ChangePassword.cshtml]:
```suggestion
                                    <input type="password" class="form-control" name="ConfirmNewPassword" id="ConfirmNewPassword" required maxlength="@AbpUserBase.MaxPlainPasswordLength">
```

File: aspnet-core/src/AbpCompanyName.AbpProjectName.Web.Mvc/Views/Users/ChangePassword.cshtml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                <h1>@L("UpdatePassword")</h1>
+            </div>
+        </div>
+    </div>
+</section>
+<section class="content">
+    <div class="container-fluid">
+        <div class="row">
+            <div class="col-12">
+                <form asp-route="ChangePassword" id="ChangePassword" method="post" autocomplete="off" class="form-horizontal">
```
Comment (demirmusa) [inline aspnet-core/src/AbpCompanyName.AbpProjectName.Web.Mvc/Views/Users/ChangePassword.cshtml]:
```suggestion
                <form id="ChangePassword" autocomplete="off" class="form-horizontal">
```

File: aspnet-core/src/AbpCompanyName.AbpProjectName.Web.Mvc/wwwroot/view-resources/Views/Users/ChangePassword.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        e.preventDefault();
+        save();
+    });
+
+    _$form.find('input').on('keypress', function (e) {
+        if (e.which === 13) {
+            e.preventDefault();
+            save();
+        }
+    });
```
Comment (demirmusa) [inline aspnet-core/src/AbpCompanyName.AbpProjectName.Web.Mvc/wwwroot/view-resources/Views/Users/ChangePassword.js]:
```suggestion
  _$form.submit(function (e) {
        e.preventDefault();
        save();
   });  
```

**Comment by iyilm4z** [comment]
@demirmusa thanks for having time, i've done fixes.

File: aspnet-core/src/AbpCompanyName.AbpProjectName.Web.Mvc/Controllers/UsersController.cs [hunk 6]
```diff
@@ -40,5 +40,7 @@ public async Task<ActionResult> EditModal(long userId)
             };
             return PartialView("_EditModal", model);
         }
+
+        public ActionResult ChangePassword() => View();
```
Comment (demirmusa) [inline aspnet-core/src/AbpCompanyName.AbpProjectName.Web.Mvc/Controllers/UsersController.cs]: We usually dont use lambda function on actionresult

**Comment by iyilm4z** [comment]
this means you want me to write boilerplate code for boilerplate?:)

**Comment by demirmusa** [comment]
> this means you want me to write boilerplate code for boilerplate?:)

Sory, but I don't understand what do you mean.

**Comment by iyilm4z** [comment]
> Sory, but I don't understand what do you mean.

[Here](https://en.wikipedia.org/wiki/Boilerplate_code) is explanantion of boilerplate code.

```csharp
public ActionResult ChangePassword () {
    return View (); 
}
```

In code block, **return** keyword inside the method and **curly braces** to wrap the method's body are boilerplate code. After we have ability to use lambda functions, that boilerplate code became _verbose_. However sure you can write some verbose code for being more explicitly as your own coding convention. Sometimes it really makes sense.

> To write boilerplate code for boilerplate.

In that sentence, second boilerplate refers to aspnetboilerplate :)

**Comment by demirmusa** [comment]
> To write boilerplate code for boilerplate.

Just tried to make it a more suitable code for the general structure.

File: aspnet-core/src/AbpCompanyName.AbpProjectName.Web.Mvc/wwwroot/view-resources/Views/Users/ChangePassword.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
+﻿(function ($) {
+    var _userService = abp.services.app.user,
+        l = abp.localization.getSource('AbpProjectName'),
+        _$form = $('#ChangePassword');
+
+    $.validator.addMethod("regex", function (value, element, regexpr) {
+        return regexpr.test(value);
+    }, l("PasswordsMustBeAtLeast8CharactersContainLowercaseUppercaseNumber"));
```
Comment (ismcagdas) [inline aspnet-core/src/AbpCompanyName.AbpProjectName.Web.Mvc/wwwroot/view-resources/Views/Users/ChangePassword.js]: Why do we force 8 length password ? We can remove the regex validation now and may implement https://docs.microsoft.com/en-us/aspnet/core/security/authentication/identity-configuration?view=aspnetcore-3.1#password in the future. @iyilm4z what do you think ?

Comment (demirmusa) [reply aspnet-core/src/AbpCompanyName.AbpProjectName.Web.Mvc/wwwroot/view-resources/Views/Users/ChangePassword.js]: Angular uses same pattern,
https://github.com/aspnetboilerplate/module-zero-core-template/blob/adfd2bdd4dd69b136dccb9606158d83f152cdcf4/angular/src/app/users/change-password/change-password.component.ts#L51

**Pull request merged**